### ISSUE TITLE
useUrlState broken with react-router@6.x

### ISSUE THREAD

**Issue by MarkLyck** [alibaba/hooks#1350]
When upgrading to react-router-dom v6.x I get the following error from use-url-state;

```
./node_modules/@ahooksjs/use-url-state/es/index.js
Attempted import error: 'useHistory' is not exported from 'react-router' (imported as 'rc').
```

I tried both of these versions:

`"@ahooksjs/use-url-state": "2.5.11",`
`"@ahooksjs/use-url-state": "^3.0.0-alpha.4",`

I get the same error on both.

react-router-dom and react-router v6, no longer has a 'useHistory export, they changed it to 'useNavigate'

**Comment by brickspert** [comment]
Hi, you should use the newest version `v3.0.0-alpha.18` , we have support react-router v6.

**Comment by MarkLyck** [comment]
@brickspert ahh thanks, I tried that version and it seems to work in the browser now, but my test environments are now broken.

I'm using Jest with the "MemoryRouter" from react-router-dom (which used to work fine).

but the new `useUrlState` seems to be incompatible with the MemoryRouter in v6?

I tried switching one of my tests from using `MemoryRouter` to `BrowserRouter` and that passed just fine.

However `MemoryRouter` is required in many of my test to set `initialEntries` to test existing URL states.

---

when using the MemoryRouter with "@ahooksjs/use-url-state" and running `setUrlState(state)` the urlState does NOT update with the new state but just keeps the default state.

**Comment by brickspert** [comment]
> @brickspert ahh thanks, I tried that version and it seems to work in the browser now, but my test environments are now broken.
> 
> I'm using Jest with the "MemoryRouter" from react-router-dom (which used to work fine).
> 
> but the new `useUrlState` seems to be incompatible with the MemoryRouter in v6?
> 
> I tried switching one of my tests from using `MemoryRouter` to `BrowserRouter` and that passed just fine.
> 
> However `MemoryRouter` is required in many of my test to set `initialEntries` to test existing URL states.
> 
> when using the MemoryRouter with "@ahooksjs/use-url-state" and running `setUrlState(state)` the urlState does NOT update with the new state but just keeps the default state.

OK, I will check it

**Comment by brickspert** [comment]
Hello~  fixed on v3.0.4
